warrior/Framework/Utils/data_Utils.py

```diff
@@ -973,6 +972,73 @@ def validate():
     return result, value
 
 
+def verify_arith_exp(expression, expected, comparison='eq'):
+    """ Verify the output of the arithmetic expression matches the expected(float comparison)
+        :Arguments:
+            1. expression: Arithmetic expression to be compared with expected.
+                This can have env & data_repo values embedded in it.
+                    Ex. expression: "10+${ENV.x}-${REPO.y}*10"
+                Expression will be evaluated based on python operator precedence
+            2. expected: Value to be compared with the expression output
+                This can be a env or data_repo or any numeral value.
+            3. comparison: Type of comparison(eq/ne/gt/ge/lt/le)
+                eq - check if both are same(equal)
+                ne - check if both are not same(not equal)
+                gt - check if expression output is greater than expected
+                ge - check if expression output is greater than or equal to expected
+                lt - check if expression output is lesser than expected
+                le - check if expression output is lesser than or equal to expected
+        :Returns:
+            1. status(boolean)
+    """
+
+    status = True
+    # Substitute env values in the expression & expected
+    expression = sub_from_env_var(expression)
+    expected = sub_from_env_var(expected)
+    # Substitute data_repo values in the expression & expected
+    expression = sub_from_data_repo(expression)
+    expected = sub_from_data_repo(expected)
+    try:
+        expression_ouput = eval(expression)
```

Comment:
Hi @venkat1077 
While eval is simple to use, it may pose some issues to us
https://stackoverflow.com/a/9558001
https://nedbatchelder.com/blog/201206/eval_really_is_dangerous.html
These gives some examples of what eval can do, since the expression is an user input, we may want to evaluate what is being input here...if warrior is running on a server, user can use this to remove files on it

The other issue I see here is the float operation
https://docs.python.org/2/tutorial/floatingpoint.html
eval("0.1+0.2") gives me 0.30000000000000004
eval("0.1+0.2==0.3") gives me False
Since this is a limitation in Python, we need to a) document and educate user b) create our own parsers for float
may also need to discuss this with the team

Reply:
Hi @kaheichan 
             Since @venkat1077 is on PTO and I need this ticket in this release. I have addressed your comments. 
1. Added an evaluation whether the user provided expression qualifies to be computed for arithmetic. This help to alleviate the concerns regarding EVAL.
2. Educated the user of the shortcomings with Python Float. 
Kindly review and if you see fit. Recommend to merge it.